eras/babbage/impl/testlib/Test/Cardano/Ledger/Babbage/TxInfoSpec.hs

```diff
@@ -147,29 +152,40 @@ txBare i o = mkBasicTx (txb i Nothing o)
 txRefInput :: forall era. (EraTx era, BabbageEraTxBody era) => TxIn -> Tx era
 txRefInput refInput = mkBasicTx (txb shelleyInput (Just refInput) shelleyOutput)
 
-hasReferenceInput :: SLanguage l -> PlutusTxInfo l -> Bool
+hasReferenceInput :: SLanguage l -> PlutusTxInfo l -> Expectation
 hasReferenceInput slang txInfo =
   case slang of
-    SPlutusV1 -> False
-    SPlutusV2 -> PV2.txInfoReferenceInputs txInfo /= mempty
-    SPlutusV3 -> PV3.txInfoReferenceInputs txInfo /= mempty
-    SPlutusV4 -> PV3.txInfoReferenceInputs txInfo /= mempty
+    SPlutusV1 -> expectationFailure "PlutusV1 does not have reference inputs"
+    SPlutusV2 -> PV2.txInfoReferenceInputs txInfo `shouldNotBe` mempty
+    SPlutusV3 -> PV3.txInfoReferenceInputs txInfo `shouldNotBe` mempty
+    SPlutusV4 -> PV3.txInfoReferenceInputs txInfo `shouldNotBe` mempty
 
-expectOneInput :: PV2.TxInInfo -> SLanguage l -> PlutusTxInfo l -> Bool
-expectOneInput i slang txInfo =
+plutusTxInInfoInputs :: forall era l. SLanguage l -> PlutusTxInfo l -> [PlutusTxInInfo era l]
```

Comment:
Would `HasCallStack` help here,  for the call to `error` below?  I guess it would have to  appear in `expectOneInput` too if we add it here